From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Max <max@localhost>
Date: Wed, 26 Aug 2026 14:00:00 +0800
Subject: [PATCH] arm64: dts: allwinner: h6: enable HDMI audio on Orange Pi 3

Wire I2S1 to the DW-HDMI codec as a simple-audio-card so ALSA can play
through the monitor speakers.

Signed-off-by: Max <max@localhost>
---
 arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts | 25 +++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts b/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts
index 75e3180c5..9b60b77f1 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts
@@ -33,6 +33,25 @@ hdmi_con_in: endpoint {
 		};
 	};
 
+
+	sound_hdmi: sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,name = "allwinner-hdmi";
+		simple-audio-card,mclk-fs = <128>;
+		simple-audio-card,frame-inversion;
+
+		simple-audio-card,codec {
+			sound-dai = <&hdmi>;
+		};
+
+		simple-audio-card,cpu {
+			sound-dai = <&i2s1>;
+			dai-tdm-slot-num = <2>;
+			dai-tdm-slot-width = <32>;
+		};
+	};
+
 	ext_osc32k: ext_osc32k_clk {
 		#clock-cells = <0>;
 		compatible = "fixed-clock";
@@ -132,6 +151,11 @@ &gpu {
 };
 
 &hdmi {
+	#sound-dai-cells = <0>;
+	status = "okay";
+};
+
+&i2s1 {
 	status = "okay";
 };
 
-- 
2.35.3
